### PR TITLE
Fix 32 byte character limit on obfuscation.

### LINKED ISSUE #301: Scrubber throws `IndexOutOfBoundsException` for token matches greater than 32 characters
[1,3,0,1]
Using token-based scrubbing, if any token matches a string of greater than 32 characters, an exception is thrown and the target file gets omitted entirely from final scrubbed bundle. This seems to be down to the fact that replacement tokens are derived from a generated `UUID`, which is 32 bytes in length.

https://github.com/elastic/support-diagnostics/blob/master/src/main/java/com/elastic/support/scrub/ScrubProcessor.java#L162-L171

Simplest solution here would probably be just to pad the string with some arbitrary character, such as `=` as is convention for many other encoding schemes.

### PR DESCRIPTION
**Pull request by nemonster**
Fixed issue where Strings longer than 32 characters generated an error during token replacement.
Added option to obfuscate a single file rather than a full diagnostic archive.
Closes #301 , #302

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,7 +58,7 @@ As a first step the diagnostic will check the Github repo for the current releas
 
 ## Using With Shield/Security
```
Comment (pickypg) [inline README.md]:
```suggestion
## Using With Security
```

File: README.md [hunk 2]
```diff
@@ -59,7 +58,7 @@ As a first step the diagnostic will check the Github repo for the current releas
...
 * a truststore does not need to be specified - it's assumed you are running this against a node that you set up and if you didn't trust it you wouldn't be running this.
-* When using Shield authentication, do not specify a password and the `-p` option.  Using the `-p` option will bring up a prompt for you to type an obfuscated value that will not be displayed on the command history.
+* When using security authentication, do not specify a password and the `-p` option.  Using the `-p` option will bring up a prompt for you to type an obfuscated value that will not be displayed on the command history.
```
Comment (pickypg) [inline README.md]:
```suggestion
* When using authentication, do not specify both a password and the `-p` option.  Using the `-p` option will bring up a prompt for you to type an obfuscated value that will not be displayed on the command history.
```

File: src/main/java/com/elastic/support/scrub/ScrubProcessor.java [hunk 3]
```diff
@@ -161,10 +158,21 @@ public String processIpv6Addresses(String input){
...
+      StringBuffer newToken = new StringBuffer();
+
+      int passes = 1;
+      if(len > 32){
+         passes = (len/32) + 1;
+      }
+
+      for(int i = 0; i < passes; i++){
+         newToken.append(
+                 ("" + (UUID.nameUUIDFromBytes(token.getBytes())).toString())
```
Comment (pickypg) [inline src/main/java/com/elastic/support/scrub/ScrubProcessor.java]:
```suggestion
                 UUID.nameUUIDFromBytes(token.getBytes()).toString()
```

File: src/main/java/com/elastic/support/scrub/ScrubProcessor.java [hunk 4]
```diff
@@ -161,10 +158,21 @@ public String processIpv6Addresses(String input){
 
    public String generateReplacementToken(String token, int len){
 
-      String replacement =
-         ("" + (UUID.nameUUIDFromBytes(token.getBytes())).toString())
-         .replaceAll("-", "")
-         .substring(0, len);
+      StringBuffer newToken = new StringBuffer();
```
Comment (pickypg) [inline src/main/java/com/elastic/support/scrub/ScrubProcessor.java]:
```suggestion
      StringBuilder newToken = new StringBuilder();
```

File: src/main/java/com/elastic/support/util/ArchiveUtils.java [hunk 5]
```diff
@@ -110,6 +109,13 @@ public void extractDiagnosticArchive(String sourceInput, String targetDir, boole
             String name = tae.getName();
             name = name.replace(baseArchivePath, "");
 
+            if(  name.contains("gc.log.") ||
+                    (name.contains(".log") && !name.equalsIgnoreCase("diagnostics.log") ) ||
```
Comment (pickypg) [inline src/main/java/com/elastic/support/util/ArchiveUtils.java]: Will this still work on 7.x JSON-based logs?

Comment (nemonster) [reply src/main/java/com/elastic/support/util/ArchiveUtils.java]: No, I haven't started on that yet. They won't be collected as of now. When I start on that I will need to address that here.

**Pull request merged**